### PR TITLE
Refactor socket connection 

### PR DESCRIPTION
**Pull request by dlhxu**
## Description
The team was facing an issue with passing around socket objects in the process of routing and redirecting. 

This change gives all components access to same socket connection via importing.

This change eliminates the need for lobby to render different pages inside of it.

## Testing
Manually tested and confirmed original functionality has not been broken

### CODE REVIEW

File: wewatch-client/src/Socket.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+// change to http://localhost:5000 for local development
+const ENDPOINT  = 'http://localhost:5000';
+
+// change to https://wewatch-server.herokuapp.com/ for production deployment
+// const ENDPOINT = 'https://wewatch-server.herokuapp.com/';
+
+let socket = io(ENDPOINT)
+
+export default socket
```
Comment (jeffreyquiambao) [inline wewatch-client/src/Socket.js:11]: awesome! this will definitely help making things easier

**Review by jeffreyquiambao — APPROVED** [review]

File: wewatch-client/src/Socket.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import io from "socket.io-client";
+
+// change to http://localhost:5000 for local development
+const ENDPOINT  = 'http://localhost:5000';
+
+// change to https://wewatch-server.herokuapp.com/ for production deployment
+// const ENDPOINT = 'https://wewatch-server.herokuapp.com/';
+
+let socket = io(ENDPOINT)
```
Comment (rosakng) [inline wewatch-client/src/Socket.js:9]: this was such a huge blocker for us! thanks for finding a solution quickly! 👏

**Review by rosakng — APPROVED** [review]

**Review by billxsheng — APPROVED** [review]
Great work!

**Review by varoongupta — APPROVED** [review]
Looks good!

**Pull request merged**